pkg/datastore/pubsub/avro/avro.go

```diff
@@ -232,7 +298,74 @@ func (a *Avro) IsSet() bool {
 	return true
 }
 
+// Deprecated: This method will be removed in future instead use Client
 // CommitOffset marks a particular offset on a specific partition as Read.
 func (a *Avro) CommitOffset(offsets pubsub.TopicPartition) {
 	a.pubSub.CommitOffset(offsets)
 }
+
+// Serialize method is used to serialize the value with a defined a schema and attaches
+// the schema Id with the value itself(wire format)
+/* Note: the Confluent schema registry has special requirements for the Avro serialization rules,
+not only need to serialize the specific content, but also attach the Schema ID and Magic Byte.*/
+// Ref: https://docs.confluent.io/current/schema-registry/serializer-formatter.html#wire-format
+func (a *Client) Serialize(value interface{}) ([]byte, error) {
+	var binaryMsg []byte
+
+	n := 4
+
+	// Confluent serialization format version number; currently always 0.
+	binaryMsg = append(binaryMsg, byte(0))
+
+	// 4-byte schema ID as returned by Schema Registry
+	binarySchemaID := make([]byte, n)
+
+	binary.BigEndian.PutUint32(binarySchemaID, uint32(a.schemaID))
+
+	binaryMsg = append(binaryMsg, binarySchemaID...)
+
+	// Missing schema will generate panic
+	if a.schema == nil {
+		return nil, &errors.Response{Code: "Missing schema", Reason: "Avro is initialized without schema"}
+	}
+
+	binaryValue, err := avro.Marshal(a.schema, value)
+	if err != nil {
+		return nil, &errors.Response{Reason: "Error in Marshaling the schema"}
+	}
+
+	// Avro serialized data in Avro's binary encoding
+	binaryMsg = append(binaryMsg, binaryValue...)
+
+	return binaryMsg, nil
+}
+
+// Deserialize method deserialize the event with checking the avro compatibility
+func (a *Client) Deserialize(msg *pubsub.Message) (finalMsg *pubsub.Message, err error) {
+	value := []byte(msg.Value)
+	schemaID := binary.BigEndian.Uint32(value[1:5])
```

Comment:
By doing this, it makes the code more readable and extensible as in future if the value is required for some other use-case, it can be used directly without extracting the bytes again and again.